cloud/linode/loadbalancers.go

```diff
@@ -243,15 +308,44 @@ func (l *loadbalancers) UpdateLoadBalancer(ctx context.Context, clusterName stri
 
 		rebuildOpts.Nodes = newNBNodes
 
-		if _, err = l.client.RebuildNodeBalancerConfig(ctx, lb.ID, currentNBCfg.ID, rebuildOpts); err != nil {
+		if _, err = l.client.RebuildNodeBalancerConfig(ctx, nb.ID, currentNBCfg.ID, rebuildOpts); err != nil {
 			sentry.CaptureError(ctx, err)
 			return fmt.Errorf("[port %d] error rebuilding NodeBalancer config: %v", int(port.Port), err)
 		}
 	}
-
 	return nil
 }
 
+// UpdateLoadBalancer updates the NodeBalancer to have configs that match the Service's ports
+func (l *loadbalancers) UpdateLoadBalancer(ctx context.Context, clusterName string, service *v1.Service, nodes []*v1.Node) (err error) {
+	ctx = sentry.SetHubOnContext(ctx)
+	sentry.SetTag(ctx, "cluster_name", clusterName)
+	sentry.SetTag(ctx, "service", service.Name)
+
+	// UpdateLoadBalancer is invoked with a nil LoadBalancerStatus; we must fetch the latest
+	// status for NodeBalancer discovery.
+	serviceWithStatus := service.DeepCopy()
+	serviceWithStatus.Status.LoadBalancer, err = l.getLatestServiceLoadBalancerStatus(ctx, service)
+	if err != nil {
+		return fmt.Errorf("failed to get latest LoadBalancer status for service (%s): %s", getServiceNn(service), err)
+	}
+
+	nb, err := l.getNodeBalancerForService(ctx, serviceWithStatus)
+	if err != nil {
+		sentry.CaptureError(ctx, err)
+		return err
+	}
+
+	if !l.shouldPreserveNodeBalancer(service) {
+		if err := l.cleanupOldNodeBalancer(ctx, service); err != nil {
```

Comment:
This should only be called in the case where the `nodebalancer-id` annotation does not match the existing NodeBalancer ID. We don't want users to find that when they change a parameter for their Service, such as the port, that they get an entirely new NodeBalancer IP.

Reply:
Nevermind, the following handles this case :)

https://github.com/linode/linode-cloud-controller-manager/pull/76/files#diff-3ec7caaf647e4dffd0640f14c7410c02R153-R155